### PR TITLE
Step 5: Add clang cl pipeline for Windows

### PR DESCRIPTION
**Pull request by wrowe**
Commit Message: Step 5: Add clang cl pipeline for Windows
Additional Description:

Now that we have regenerated envoy build tools to update bazel and llvm, introduce a clang-cl pipeline based on clang-cl

Co-authored-by: Sunjay Bhatia <sunjayb@vmware.com>

Risk Level: low (new facility)
Testing: local
Docs Changes: TBD
Release Notes: TBD
Platform Specific Features: Windows clang-cl compilation
Requires https://github.com/envoyproxy/envoy/pull/13688
Fixes https://github.com/envoyproxy/envoy/issues/11974

### CODE REVIEW

**Comment by wrowe** [comment]
The most recent push dropped the step 2 patch of the bazel version number change, that's still ambiguous.

**Comment by wrowe** [comment]
/retest

since Sunjay hasn't, yet :)

**Comment by wrowe** [comment]
Now that lizan picked up step 4, going to retry a couple times with the new tooling. We did something specific in this patch which can be changed... the clang pipelines runs after the msvc pipeline, mostly to give the msvc build all the compute it needs to get finished. But depending on how gcp rbe behaves, we could invoke them in parallel.

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -409,9 +409,29 @@ stages:
               artifactName: windows.release
             condition: always()
 
-      - job: docker
+      - job: clang_cl
```
Comment (davinci26) [inline .azure-pipelines/pipelines.yml:412]: do we want to enable the deflaking script in this pipeline as well?

See where we enabled it on Windows https://github.com/envoyproxy/envoy/pull/14102/files#diff-8ca122481a2ae193003b523bf5b7a9cfa0e914be502d483b6adc1d06668c724a

Comment (sunjayBhatia) [reply .azure-pipelines/pipelines.yml:412]: +1

**Comment by wrowe** [comment]
This is being rescheduled to 1.18-dev, and will be moved to the windows-ga milestone once the interesting half... https://github.com/envoyproxy/envoy/pull/14608 has been merged, and
this commit rebased down to simply a new pipeline.

The flake and the fail noted from //test/common/network:dns_impl_test and //test/extensions/filters/network/sni_dynamic_forward_proxy:proxy_filter_integration_test
need to be added to this patch, but specific only to clang-cl builds, since msvc-cl
builds seem to be unaffected.

**Comment by wrowe** [comment]
This patch was split into 14608 (applied) and 14668 (draft).

Closing this ticket to retire the github org it was pulled from.

**Pull request closed without merging**